src/iuse.cpp

```diff
@@ -959,15 +960,26 @@ int iuse::blech(player *p, item *it, bool, const tripoint& )
             return 0;
         }
     } else { //Assume that if a blech consumable isn't a drink, it will be eaten.
-        if (!p->query_yn(_("This looks unhealthy, sure you want to eat it?"))) {
+        if (!p->query_yn( _("This looks unhealthy, sure you want to eat it?"))) {
             return 0;
         }
     }
-    p->add_msg_if_player(m_bad, _("Blech, that burns your throat!"));
-    p->mod_pain(rng(32, 64));
-    p->add_effect( effect_poison, 600);
-    p->apply_damage(nullptr, bp_torso, rng(4, 12));
-    p->vomit();
+
+    if( it->has_flag( "ACID" ) && ( p->has_trait( trait_ACIDPROOF ) || p->has_trait( trait_ACIDBLOOD ))) {
+        p->add_msg_if_player(m_bad, _("Blech, that tastes gross!"));
+        //partially reverse the harmful values of drinking this acid.
+        double multiplier = -0.5;
+        p->mod_hunger( -p->nutrition_for( *it ) * multiplier );
+        p->mod_thirst( -it->type->comestible->quench * multiplier );
+        p->mod_healthy_mod( it->type->comestible->healthy * multiplier, it->type->comestible->healthy * multiplier );
+        p->add_morale( MORALE_FOOD_BAD, it->type->comestible->fun * multiplier, 60, 60, 30, false, it->type );
```

Comment:
There is still no reason to ever drink this, so this code will not be invoked.
You could add something useful here, like purging some parasites, adding minor nutrition (acid metabolism) or something like that.
Note that acid immunity is 100% - it's not "you get dissolved slower", but "acid does literally nothing to you".

Don't add morale like this. Having morale bonus and penalties mix in one type may get wonky due to rules on stacking limits.

Reply:
In regards to the morale, this what the plantblech code also did. So I have no idea how to do it better. Esp as this it is in place to get around the bad morale penalities that get evoked from just consuming the object. (which is done somewhere else in the code).

Reply:
I looked at simply making it call the anti parasitic function, as a bonus for drinking the acid, but that function can do weird things as it can't be used underwater. (For some strange reason, it is part of the few iuse functions that you cant use underwater, bit inconsistent it seems to me).

So now im thinking of just making it give 20 quench.

In regards to the morale penalty: 
I tried p->rem_morale( MORALE_FOOD_BAD, it->type ); but that only removes the past morale additions of drinking the acids, and then some other function adds the negative morale of drinking the acid again.